/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax="proto2";

import "TouchEventData.proto";
import "ButtonEventsData.proto";
import "AbsoluteInputEventsData.proto";
import "RelativeInputEventsData.proto";

package aasdk.proto.messages;

message InputEventIndication
{
    required uint64 timestamp = 1;
    optional int32 disp_channel = 2;
    optional data.TouchEvent touch_event = 3;
    optional data.ButtonEvents button_event = 4;
    optional data.AbsoluteInputEvents absolute_input_event = 5;
    optional data.RelativeInputEvents relative_input_event = 6;
}
